core/contracts/financial-templates/implementation/PricelessPositionManager.sol

```diff
@@ -214,12 +215,9 @@ contract PricelessPositionManager is FeePayer, AdministrateeInterface {
         PositionData storage positionData = _getPositionData(msg.sender);
         require(positionData.requestPassTimestamp == 0);
 
-        _removeCollateral(positionData.rawCollateral, collateralAmount);
+        // Decrement the sponsor's collateral and global collateral amounts.
+        amountWithdrawn = _decrementCollateralBalances(positionData, collateralAmount);
         require(_checkPositionCollateralization(positionData));
```

Comment:
@nicholaspai @mrice32 and @ptare I need some input here: Before we decremented the positions `rawCollateral` by the `collateralAmount` then did the check for correct collateralization. After this we decremented the global collateral. This ordering meant that the check compared the decremented positions collateral against the non-decremented global collateral.

In my change both the position and global collateral are decremented before doing the check. 

Specifically this becomes an issue if there is only one sponsor within the contract as the global `rawTotalPositionCollateral ` is then equal to the position's `rawCollateral`. This means that a single sponsor withdrawal _can_ bring the total contract collateralization ratio below the bottom CR of the contract. This can be see in this specific unit test failing.

To deal with this I tried moving this require into the `_decrementCollateralBalances` such that All decrements check the `_checkPositionCollateralization` assertion before decrementing global collateral. However I'm not sure if this is sensical in this case.

The other problem with performing this change is that previously the amount withdrawn was the amount by which the `rawTotalPositionCollateral` was decremented (see comment below). This is now no longer the case with this change.

So my questions to you'll are: 
1) do we want to refactor this removal logic into a seperate function, given that it is consistently implemented in a few places (sometimes use the `rawTotalPositionCollateral` as the `amountWithdrawn`, sometimes use the position size changed as the `amountWithdrawn` etc).
2) What should the correct ordering of the two removals be in the context of this require?


If we choose to not do 1 then do we want to scrap this change in general? However I do feel that in all other cases other than this function it is an improvement. LMKWYT.

Reply:
>To deal with this I tried moving this require into the _decrementCollateralBalances such that All decrements check the _checkPositionCollateralization assertion before decrementing global collateral. However I'm not sure if this is sensical in this case.
I agree this isn't something we should do every time we remove collateral from the contract, for example in `withdrawPassedRequest` or `redeem`.

>do we want to refactor this removal logic into a seperate function, given that it is consistently implemented in a few places (sometimes use the rawTotalPositionCollateral as the amountWithdrawn, sometimes use the position size changed as the amountWithdrawn etc).
I think this might be the best option: have 2 separate `_decrementCollateral` methods, one performs the GCR check and one does not.

Reply:
Alright @nicholaspai I implemented this change. take a look and let me know what you think.

Reply:
This looks good, do all tests pass without modification now?